src/network/types.rs

```diff
@@ -89,6 +89,13 @@ pub struct NetworkOptions {
     /// Custom DNS servers for aardvark-dns.
     #[serde(rename = "dns_servers")]
     pub dns_servers: Option<Vec<IpAddr>>,
+
+    /// The container hostname.
+    #[serde(
+        rename = "container_hostname",
+        default = "NetworkOptions::default_hostname"
+    )]
+    pub container_hostname: String,
```

Comment:
Can you move this field up below container_name as they are rather similar.

Second, instead of defining a default function you could use `#[serde(default)]`. However given this is optional I rather instead wrap the type in `Option<>` to make this more clear. Then on the caller side you can do
`container_hostname.unwrap_or("")`

Reply:
> Can you move this field up below container_name as they are rather similar.

Done.  Decades of "C" development compel me to put new fields at the end of structures. :)




> Second, instead of defining a default function you could use `#[serde(default)]`. However given this is optional I rather instead wrap the type in `Option<>` to make this more clear. Then on the caller side you can do `container_hostname.unwrap_or("")`

This is my first practical exposure to Rust so I _think_ I did this correctly.  I did test it with current podman to make sure it didn't choke if container_hostname wasn't present in the JSON.